src/paths/dijkstra.c

```diff
@@ -1019,44 +1074,87 @@ int igraph_get_all_shortest_paths_dijkstra(const igraph_t *graph,
+                            IGRAPH_CHECK(igraph_vector_copy(path, parent_path_edge));
+                            IGRAPH_CHECK(igraph_vector_ptr_push_back(edges, path));
+                            IGRAPH_FINALLY_CLEAN(1);  /* ownership of path passed to vertices */
+                            IGRAPH_CHECK(igraph_vector_push_back(path, parent_edge));
+                        } else {
+                            IGRAPH_CHECK(igraph_vector_init(path, 1));
+                            IGRAPH_CHECK(igraph_vector_ptr_push_back(edges, path));
+                            IGRAPH_FINALLY_CLEAN(1);  /* ownership of path passed to vertices */
+                            VECTOR(*path)[0] = parent_edge;
+                        }
+                    }
                 }
             }
         }
-
-        /* remove the destructor from the path vector */
-        igraph_vector_ptr_set_item_destructor(res, 0);
+        if (!free_vertices) {
+            /* remove the destructor from the path vector */
+            igraph_vector_ptr_set_item_destructor(vertices, 0);
+        }
 
         /* free those paths from the result vector which we won't need */
-        n = igraph_vector_ptr_size(res);
+        n = igraph_vector_ptr_size(vertices);
         j = 0;
         for (i = 0; i < n; i++) {
             igraph_real_t tmp;
-            path = (igraph_vector_t*)VECTOR(*res)[i];
+            path = (igraph_vector_t*)VECTOR(*vertices)[i];
             tmp = igraph_vector_tail(path);
             if (is_target[(long int)tmp] == 1) {
                 /* we need this path, keep it */
-                VECTOR(*res)[j] = path;
+                if (!free_vertices) {
+                    VECTOR(*vertices)[j] = path;
+                }
+                if (edges) {
+                    path = (igraph_vector_t*)VECTOR(*edges)[i];
+                    VECTOR(*edges)[j] = path;
+                }
                 j++;
             } else {
                 /* we don't need this path, free it */
-                igraph_vector_destroy(path); free(path);
+                if (!free_vertices) {
+                    igraph_vector_destroy(path); free(path);
+                }
+                if (edges) {
+                    path = (igraph_vector_t*)VECTOR(*edges)[i];
+                    igraph_vector_destroy(path); free(path);
+                }
             }
         }
-        IGRAPH_CHECK(igraph_vector_ptr_resize(res, j));
-
-        /* sort the paths by the target vertices */
-        igraph_vector_ptr_sort(res, igraph_i_vector_tail_cmp);
+        if (!free_vertices){
+            IGRAPH_CHECK(igraph_vector_ptr_resize(vertices, j));
+            /* sort the paths by the target vertices */
+            igraph_vector_ptr_sort(vertices, igraph_i_vector_tail_cmp);
+        }
+        
+        if (edges){
+            IGRAPH_CHECK(igraph_vector_ptr_resize(edges, j));
+            igraph_vector_ptr_sort(edges, igraph_i_vector_tail_cmp);
+        }
```

Comment:
Note to self: We have a problem here: in the preceding lines, the vertex paths are sorted based on their target vertex. The edge path list should be ordered identically to the vertex path list.